core/src/main/java/org/elasticsearch/common/xcontent/support/AbstractXContentParser.java

```diff
@@ -289,7 +289,7 @@ public BytesRef utf8BytesOrNull() throws IOException {
             token = parser.nextToken();
         }
         ArrayList<Object> list = new ArrayList<>();
-        for (; token != XContentParser.Token.END_ARRAY; token = parser.nextToken()) {
+        for (; token != null && token.isValue(); token = parser.nextToken()) {
```

Comment:
maybe this should be `token != null && (token.isValue() || token == VALUE_NULL)`?